### PR TITLE
Level order traversal refactoring, added unit test

### PR DESCRIPTION
**Pull request by albina-astr**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/test/java/com/thealgorithms/datastructures/trees/ZigzagTraversalTest.java [hunk 1]
```diff
@@ -22,9 +22,33 @@ public void testSingleNodeTree() {
         assertEquals(List.of(List.of(50)), ZigzagTraversal.traverse(root));
     }
 
+    /*
```
Comment (albina-astr) [inline src/test/java/com/thealgorithms/datastructures/trees/ZigzagTraversalTest.java:25]: Rewrote tests a little bit to make them more readable.

File: src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java [hunk 2]
```diff
@@ -1,58 +1,46 @@
...
+/* Class to print Level Order Traversal */
+public class LevelOrderTraversal {
 
-    /* Compute the "height" of a tree -- the number of
-  nodes along the longest path from the root node
-  down to the farthest leaf node.*/
-    int height(Node root) {
+    /* Given a binary tree. Print its nodes in level order
+  using array for implementing queue  */
+    static List<List<Integer>> traverse(BinaryTree.Node root) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java:13]: Traversal is just a sequence of nodes, why does it return `List<List<Integer>>` and not `List<Integer>`?

File: src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java [hunk 3]
```diff
@@ -1,58 +1,46 @@
...
+        List<List<Integer>> result = new ArrayList<>();
+
+        Queue<BinaryTree.Node> q = new LinkedList<>();
+        q.add(root);
+        while (!q.isEmpty()) {
+            int nodesOnLevel = q.size();
+            List<Integer> level = new LinkedList<>();
+            /* poll() removes the present head.
+               For more information on poll() visit
+               http://www.tutorialspoint.com/java/util/linkedlist_poll.htm */
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java:27]: Please remove trivial comments, they just duplicate what's already clear from the source code

File: src/test/java/com/thealgorithms/datastructures/trees/ZigzagTraversalTest.java [hunk 4]
```diff
@@ -22,9 +22,33 @@ public void testSingleNodeTree() {
...
+        final BinaryTree.Node root = TreeTestUtils.createTree(new Integer[]{1, 2, 3, 4, 5, 6, 7});
+        assertEquals(List.of(List.of(1), List.of(3, 2), List.of(4, 5, 6, 7)), ZigzagTraversal.traverse(root));
+    }
+
+    /*
+         1
+        / \
+       2   3
+      /\   /\
+     4   56  7
```
Comment (siriak) [inline src/test/java/com/thealgorithms/datastructures/trees/ZigzagTraversalTest.java:43]:
```suggestion
     4  5 6  7
```

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java (hunk 2 above)
Comment (albina-astr) [reply src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java:13]: Actually, it's not just a sequence of nodes. Here we return ```List<List<Integer>>``` to store nodes level by level. Also, this approach can help to print the nodes level by level.

File: src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java (hunk 3 above)
Comment (albina-astr) [reply src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java:27]: The original code and the comments aren't mine, I'll fix this.

File: src/test/java/com/thealgorithms/datastructures/trees/ZigzagTraversalTest.java (hunk 4 above)
Comment (albina-astr) [reply src/test/java/com/thealgorithms/datastructures/trees/ZigzagTraversalTest.java]: fixed

**Comment by albina-astr** [comment]
@siriak Hi! I've fixed this PR.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**